### PR TITLE
Tpbot/ Testable Movement Service

### PR DESCRIPTION
**Pull request by Mecharyry**
#### Problem
In order for the `Bot` to work completely in the background, we need to move the `MovementService` to be a bound service that is destroyed in `onDestroy`. Unfortunately, I don't know how the `MovementService` works 😆 

#### Solution
Pull out testable collaborators for the `MovementService` and test them! In a follow-up PR we should be able to move everything to be in a bound service where the `BotService` forwards commands it receives to this `Service` 😄

### CODE REVIEW

File: TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/SerialPortCreator.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.novoda.tpbot.bot;
+
+import android.hardware.usb.UsbDevice;
+import android.hardware.usb.UsbDeviceConnection;
+
+import com.felhr.usbserial.UsbSerialDevice;
+
+class SerialPortCreator {
+
+    UsbSerialDevice create(UsbDevice usbDevice, UsbDeviceConnection connection) {
```
Comment (Mecharyry) [inline TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/SerialPortCreator.java]: Can't mock a static so needed to wrap this 😢

File: TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/MovementServiceIntentHandler.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import android.hardware.usb.UsbManager;
+
+import com.novoda.notils.exception.DeveloperError;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.novoda.tpbot.bot.AndroidMovementService.ACTION_USB_PERMISSION;
+
+class MovementServiceIntentHandler {
```
Comment (Mecharyry) [inline TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/MovementServiceIntentHandler.java]: Ah snap, I was meant to test this 😞

Comment (Mecharyry) [inline TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/MovementServiceIntentHandler.java]: Addressed in 	b0eb46b

File: TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/AndroidMovementService.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        createBroadcastReceiver();
+        startConnection();
+        return START_STICKY;
+    }
+
+    private void createBroadcastReceiver() {
```
Comment (danybony) [inline TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/AndroidMovementService.java]: More like `registerBroadcastReceiver()`

File: TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/SerialPortMonitor.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            serialPort.setStopBits(UsbSerialInterface.STOP_BITS_1);
+            serialPort.setParity(UsbSerialInterface.PARITY_NONE);
+            serialPort.setFlowControl(UsbSerialInterface.FLOW_CONTROL_OFF);
+            serialPort.read(onDataReceivedListener);
+            return true;
+        } else {
+            stopMonitoring();
+            return false;
+        }
+
```
Comment (danybony) [inline TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/SerialPortMonitor.java]: Useless empty line

File: TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/MovementServiceIntentHandler.java (hunk 2 above)
Comment (danybony) [inline TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/MovementServiceIntentHandler.java]: 💯

**Pull request merged**